service/portability_service_template.json

```diff
@@ -0,0 +1,614 @@
+            }
+        },
+        "TestsGetStatusFunction": {
+            "Type": "AWS::Lambda::Function",
+            "Properties": {
+                "Description": "Handles GET to /portability_tests/{test_id}/status by retrieving test status from registered environments.",
+                "Code": {
+                    "S3Bucket": {"Ref": "LambdaCodeBucket"},
+                    "S3Key": {"Ref": "LambdaCodeKey"}
+                },
+                "Environment": {
+                    "Variables": {
+                        "ENVIRONMENTS_TABLE": {"Ref": "EnvironmentsTable"},
+                        "TESTS_TABLE": {"Ref": "TestsTable"},
+                        "REGION": {"Ref": "AWS::Region"}
+                    }
+                },
+                "FunctionName": {"Fn::Join": ["-", [{"Ref": "AWS::StackName"}, "tests-get-status-fn"]]},
+                "Handler": "portability_service.portability_tests_get_status",
+                "Role": {"Fn::GetAtt": ["LambdaRole", "Arn"]},
+                "Runtime": "python3.6",
+                "Timeout": 30
+            }
+        },
+        "PortabilityApi": {
+            "Type": "AWS::ApiGateway::RestApi",
+            "Properties": {
+                "Name": {"Fn::Join": ["-", [{"Ref": "AWS::StackName"}, "rest-api"]]},
+                "Description": "Accepts requests for workflow portability tests.",
+                "Body": {
+                    "swagger": "2.0",
+                    "info": {
+                        "title": "Portability Service API",
+                        "version": "0.0.1"
+                    },
+                    "securityDefinitions": {
+                        "api_key": {
+                            "type": "apiKey",
+                            "name": "x-api-key",
+                            "in": "header"
+                        }
+                    },
+                    "schemes": ["https"],
+                    "consumes": ["application/json"],
+                    "produces": ["application/json"],
+                    "definitions": {
+                        "empty": {
+                            "type": "object"
+                        },
+                        "error": {
+                            "type": "object",
+                            "properties": {
+                                "message": {
+                                    "type": "string"
+                                },
+                                "type": {
+                                    "type": "string"
+                                },
+                                "request-id": {
+                                    "type": "string"
+                                }
+                            }
+                        },
+                        "fullenvironmentdescription": {
+                            "type": "object",
+                            "properties": {
+                                "name": {
```

Comment:
Can you add a `definition` for each of the properties under API definitions?

Reply:
I think I am allowed to give it a `title`. I will try that.

Reply:
I thought a `description` key was allowed here but it's possible I was looking at a different swagger version

Reply:
Maybe in swagger 3.0? There seem to be a lot more `description` fields in there, but I don't think AWS lets you use that with cloudformation yet.